# Cheat Sheet 8: Hypothesis Testing and the Wald Test

_MMA 860 – Management of Data_

## What this covers

- The structure of a hypothesis test in a regression context
- The Wald test for joint linear restrictions
- Interpreting p-values correctly
- Common mistakes in framing null hypotheses

## Working with AI

AI tools are surprisingly good at writing the syntax for a Wald test (the string format trips most students up). They are less good at framing the null hypothesis correctly – that is your job.

Try this prompt when you are stuck: *"I want to test whether [business question]. What's the right null hypothesis to set up in a Wald test, and what would the alternative be?"* Verify with the textbook.

## Reminder on what a p-value is – and is not

A p-value is the probability of observing data at least as extreme as what we saw, *assuming the null hypothesis is true*. That is it.

It is not the probability that the null is true. It is not 1 minus the probability your model is right. It is not a measure of effect size. Students lose marks every year on this. Be precise.

# Import data and run regression

The data can be found in the structural change data tab in the Cheat_Sheet_Hypothesis_Testing_V1_0 spreadsheet.

In [6]:
#Import the required libraries
import pandas as pd
import os.path as osp
import statsmodels.api as sm
from statsmodels.formula.api import ols

#Build the path for the data file
data_path = osp.join(
    osp.curdir,'Data','Cheat_Sheet_Hypothesis_Testing_V1_0.xlsx')

#Use the read_excel function to pull data from the Hypothesis Testing Sheet
data = pd.read_excel(
    data_path,sheet_name=0,index_col='Obs')

#Build our model using statsmodels
model = ols('Y ~ X1 + X2 + X3 + X4 + X5',data).fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      Y   R-squared:                       0.867
Model:                            OLS   Adj. R-squared:                  0.863
Method:                 Least Squares   F-statistic:                     251.9
Date:                Thu, 16 Jul 2026   Prob (F-statistic):           8.52e-83
Time:                        18:03:09   Log-Likelihood:                -1029.1
No. Observations:                 200   AIC:                             2070.
Df Residuals:                     194   BIC:                             2090.
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    114.4941     14.438      7.930      0.000      86.019     142.969
X1             9.6945      5.990      1.619      0.107      -2.118      21.507
X2             0.0037      0.000      8.059      0.000       0.003       0.005
X3             0.0224      0.011      2.102      0.037       0.001       0.043
X4             0.1051      0.003     33.691      0.000       0.099       0.111
X5            -9.9705      6.454     -1.545      0.124     -22.699       2.758
==============================================================================
Omnibus:                        4.399   Durbin-Watson:                   1.280
Prob(Omnibus):                  0.111   Jarque-Bera (JB):                3.228
Skew:                           0.171   Prob(JB):                        0.199
Kurtosis:                       2.480   Cond. No.                     6.33e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 6.33e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

# Performing Hypothesis Testing using the Wald Test

Using the following arguments, we will test and see whether $X_1$, $X_2$, and $X_3$ have a coefficient of zero in front of them. In other words, we are testing the correllation.

In [9]:
'''
We put our null hypothesis in the following string format. Here we are going
to test whether X1, X2, and X3 have a 0 coefficient in front of them.
'''
hypothesis = '(X1=0, X2=0, X3=0)'

In [10]:
#Pass the hypothesis to the Wald_Test
print(model.wald_test(hypothesis))

<F test: F=array([[25.19193307]]), p=8.273060507484384e-14, df_denom=194, df_num=3>


The p-value above is what we are concerned with the most. You should have gotten a value of 8.27306050734528e-14 (the 'e' here means $\times 10 \text{ to the power of}$). For example, 8.27e-14 is $8.27\times 10^{-14}$ (very small!). If our p-value is less than $0.05$, which it clearly is here, then we **reject the null hypothesis**. This tells us that there is, to great confidence, some correllation between the three parameters and our $Y$ value. Let's try another example where we test the correllation of just $X_1$ and $X_5$.

In [ ]:
hypothesis_2 = '(X1=30)'
print(model.wald_test(hypothesis_2))

Here, our p-value is much higher, about $0.09$. This p-value is not less than $0.05$, so in this case, we **do not reject the null hypothesis**. Note, this does not mean that we have confirmed $X_1$ and $X_5$ are not correllated with our $Y$ value, we simply found that there is not enough evidence to suggest that they are.

# General Notes

- All of these restrictions are tested simultaneously as part of the null hypothesis.
- You can test many null hypotheses using the linearHypothesis command.
- There are other tests (for heteroskedasticity, specification error, etc.) that cannot be tested using this approach.

## Try it yourself

1. Test whether X4 alone is significantly different from zero. Compare to the t-test in the summary.
2. Test whether the sum of X1 and X2 coefficients equals 1. (Syntax: `'X1 + X2 = 1'`.)
3. **AI exercise.** Describe a business question in plain language to your AI tool. Ask it to translate that into a Wald test specification. Critique the result.